src/ucp/wireup/address.c

```diff
@@ -320,13 +320,18 @@ ucp_address_get_device(ucp_context_h context, ucp_rsc_index_t rsc_index,
     return dev;
 }
 
-static size_t ucp_address_packed_value_size(size_t value, size_t max_value,
-                                            ucp_object_version_t addr_version)
+static ssize_t ucp_address_packed_value_size(size_t value, size_t max_value,
+                                             ucp_object_version_t addr_version)
 {
     if (addr_version == UCP_OBJECT_VERSION_V1) {
         /* Address version 1 does not support value extension */
-        ucs_assertv_always(value <= max_value, "value %zu, max_value %zu",
-                           value, max_value);
+        if (value > max_value) {
+            ucs_debug("value %zu > max_value %zu, use UCX_ADDRESS_VERSION=v2 to"
+                      " allow packing more addresses",
+                      value, max_value);
+            return -1;
```

Comment:
Can we return the UCS error status here (they are negative)? instead of -1, return UCS_ERR_UNSUPPORTED for example
it's similar to CUT bcopy API return value, for example

Reply:
fixed